lib/src/over_react_test/custom_matchers.dart

```diff
@@ -337,3 +339,203 @@ Matcher throwsPropError_Combination(String propName, String prop2Name, [String m
       ))
   ));
 }
+
+/// A log matcher that ensures that expected log values are present.
+///
+/// In its core, the matcher compares two lists. The added utility is that by
+/// passing in a callback, the matcher will run the function and record the logs,
+/// return them in a list, and use that list as the actual value to compare
+/// against the expected.
+///
+/// Related: [recordConsoleLogs]
+class _LogMatcher extends Matcher {
+  _LogMatcher.logsPropTypeWarning(/*String || Contains*/ this._expected, {this.shouldEnforceLogCount = false})
+      : _expectsSingleMatch = true, _expectsMatch = true;
+
+  _LogMatcher.logsPropTypeWarnings(/*String || Contains*/ this._expected, {this.shouldEnforceLogCount = false})
+      : _expectsSingleMatch = false, _expectsMatch = true;
+
+  _LogMatcher.logsNoPropTypeWarnings({this.shouldEnforceLogCount = false})
+      : _expected = contains('Failed prop type:'), _expectsSingleMatch = false, _expectsMatch = false;
```

Comment:
the `_expected = contains('Failed prop type:')` feels weird to me because the actual expected is `[]` right? i get that this is like a way to like trick it because then you set `expects[singleMatch]` false but couldn't we add some logic that for `_expected.isEmpty` or something and then handle it that way? because for example couldn't a user put in `logsPropTypeWarnings([])`... would that fail?

Reply:
I see what you're saying! Because it's an empty list, you'd expect that the matcher to say that there shouldn't be any propType warnings. And we could build it that way, but under the hood we would still need a place where we look through all of the actual making sure there are no `propType` error matches, making it essentially the same logic as `logsPropTypeWarnings`. If we want to build in that logic to `logsPropTypeWarnings`, we can do that!

That being said, I do feel like it's a misuse of `logsPropTypeWarnings` because we have a matcher for the use case of verifying that there are no `propTypes` warnings, so I feel a little more comfortable making an assumption with `logsPropTypeWarnings` that they're looking for multiple warnings. In any case, you definitely made a really good catch at a gaping whole in the matcher. No matter what, it doesn't handle an empty list / null well 😄

Reply:
i mean i'd be fine with it being dynamic and not having a `logsPropTypeWarning` lol but if we do want both then the typing makes sense even though i hate it and would prefer it be dynamic lol :P

Reply:
Oh for sure, 100% the typing was a great idea. My stream of consciousness was more of regarding what happens when `[]` / `null` gets passed in, which I'm now realizing was an offline conversations and not really a reply to what your message was saying haha